### PR TITLE
checkpoint 04 - Erin Dawson

### PR DESCRIPTION
**Pull request by edawson1980**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,9 @@ Demonstrate calling the method, passing in "young prince" as the argument.
 Write your code here...
 
 ```ruby
-# code here
+def offer_rose(person)
+  puts "Would you take this rose, #{offer_rose}, in exchange for giving an old beggar woman shelter from the bitter cold?"
+end
```
Comment (jsm13) [inline README.md:34]: 👍 also want to call `offer_rose` with `"young prince"`

File: README.md [hunk 2]
```diff
@@ -54,7 +56,8 @@ Using Ruby...
 Write your code here...
 
 ```ruby
-# code here
+belle = residents.slice[1]
+guests << belle
```
Comment (jsm13) [inline README.md:60]: `slice` won't update the original array so want to call `town[:residents].slice!(1)`

also, `slice` is a little weird (in order to provide functionality that looks like array access with brackets), we can use the bracket notation `town[:residence][1]` which is sugar for calling the method with the desired index as an argument which will use parens `town[:residence].slice(1)`. In practice we always use the former

File: README.md [hunk 3]
```diff
@@ -77,7 +80,11 @@ Belle is friends with Mrs. Potts
 Write your code here...
 
 ```ruby
-# code here
+friends = ["Chip Potts", "Cogsworth", "Lumière", "Mrs. Potts"]
+friends.each do |friend|
+  puts "Belle is friends with #{friend}"
+end
```
Comment (jsm13) [inline README.md:86]: 👍

File: README.md [hunk 4]
```diff
@@ -100,7 +107,35 @@ Each lion should have...
 Create a new lion instance with the name "Simba".
 
 ```ruby
-# code here
+class Animal
+  attr_accessor(name)
+
+  def initialize
+  end
```
Comment (jsm13) [inline README.md:114]: want to set name here

File: README.md [hunk 5]
```diff
@@ -100,7 +107,35 @@ Each lion should have...
...
+    return @name
+  end
+
+  def greet
+    puts "Hi! My name is #{@name}!"
+  end
+
+end
+
+class Lion
```
Comment (jsm13) [inline README.md:126]: need to inherit from animal `class Lion < Animal`

File: README.md [hunk 6]
```diff
@@ -100,7 +107,35 @@ Each lion should have...
...
+  attr_accessor (pack)
+  super(name)
+
+  def initalize (king)
+  end
```
Comment (jsm13) [inline README.md:131]:
```ruby
def initialize(name)
  super(name)
  @king = @name == "Simba"
  @@pack << self
end

File: README.md [hunk 7]
```diff
@@ -118,7 +153,11 @@ entities (no need to draw an ERD)...
 Your answer...
 
 ```
-Replace this with your answer
+an ERD is an Entity Relationship Diagram.  it shows the relationships between different parts of an applications, which helps us to figure out exactly how we want to structure our application.
+  Genie to Lamp is a one-to-one
+  Lamp to Genie could be a one-to-many, but is probably more likely a one-to-one.
+  Person to Pet is a one-to-many.
+  Pet to Person is also a one-to many.
```
Comment (jsm13) [inline README.md:160]: 👍 these last two condition would make Pets to Persons many to many

File: README.md [hunk 8]
```diff
@@ -129,7 +168,7 @@ SQL database. If you need an example, you can use `people` and `wishes` models.
 Your answer...
 
 ```
-Replace this with your answer
+a schema is the structure for a table in our database.  a one to many relationship would be
```
Comment (jsm13) [inline README.md:171]: 👍 on what a schema is. Looking for a one to many relationship is described with a foreign key on records in the many table referring to records in the one table

**Review by jsm13 — COMMENTED** [review]
👍 good on most of these but dropped off at the end. How're you feeling with active record?

**Comment by edawson1980** [comment]
i'm feeling pretty okay with ActiveRecord conceptually.  I just need practice with keeping all the minutiae straight.

**Pull request closed without merging**